tests/quantization/ggml/test_ggml.py

```diff
@@ -694,6 +698,46 @@ def test_starcoder2_q6_k(self):
         EXPECTED_TEXT = 'def print_hello_world():\n    print("Hello World")\n\ndef print'
         self.assertEqual(tokenizer.decode(out[0], skip_special_tokens=True), EXPECTED_TEXT)
 
+    def test_mamba_weights_conversion_fp16(self):
+        original_model = AutoModelForCausalLM.from_pretrained(
+            self.mamba_original_model_id,
+            torch_dtype=torch.float16,
+        )
+
+        converted_model = AutoModelForCausalLM.from_pretrained(
+            self.mamba_model_id,
+            gguf_file=self.fp16_mamba_model_id,
+            torch_dtype=torch.float16,
+        )
+
+        converted_state_dict = converted_model.state_dict()
+        original_state_dict = original_model.state_dict()
+
+        for layer_name, original_params in original_state_dict.items():
+            if layer_name in converted_state_dict and layer_name != "lm_head.weight":
+                # quantized models do not contain "lm_head.weight" layer
```

Comment:
Hey @SunMarc! If I am not mistaken, it is because of this logic in tensor modification in https://github.com/ggerganov/llama.cpp/blob/master/convert_hf_to_gguf.py:                                                       
 `# assuming token_embd.weight is seen before output.weight
        if self._tok_embd is not None and new_name == output_name:
            if torch.equal(self._tok_embd, data_torch):
                logger.debug(f"{output_name} is equivalent to {tok_embd_name}, omitting")
                return []
        elif new_name == tok_embd_name:
            self._tok_embd = data_torch`
It just does not include `output.weight` which is `lm_head` eventually in quantization

Reply:
They removed the lm_head since it is the same of the embedding weights. This is also done in the original model. You can check the safetensors file. Still after loading the model, we should have the `lm_head`. This is linked to `tie_word_embeddings = True` in the config file. The default value is `True`. Would you like to have a look at what's happening and potentially fix the issue for starcoder2 also ? Thanks !

Reply:
Hezy @SunMarc! The problem is, that quantized model even does not have this tensor. If you check https://huggingface.co/jpodivin/mamba-2.8b-hf-GGUF?show_file_info=ggml-model-Q5_K.gguf for example or any others gguf files, in Tensor section there is no output.weight tensor. So, how could we get it for dequantization?

Reply:
Ok, got it, thanks! I'll fix that